bot/exts/fun/fun.py

```diff
@@ -153,10 +154,28 @@ async def caesarcipher_decrypt(self, ctx: Context, offset: int, *, msg: str) ->
         await self._caesar_cipher(ctx, offset, msg, left_shift=True)
 
     @commands.command()
-    async def joke(self, ctx: commands.Context, category: Literal["neutral", "chuck", "all"] = "all") -> None:
-        """Retrieves a joke of the specified `category` from the pyjokes api."""
-        joke = pyjokes.get_joke(category=category)
-        await ctx.send(joke)
+    async def joke(self, ctx: commands.Context, category: Literal["dad", "neutral", "chuck", "all"] = "all") -> None:
+        """
+        Retrieves a joke of the specified `category` from the pyjokes api.
+
+        - dad uses icanhazdadjoke.
+        - others use pyjokes.
+        """
+        if category == "dad":
+            async with aiohttp.ClientSession() as session, session.get(
+                "https://icanhazdadjoke.com",
+                headers={
+                    "Accept":"application/json",
+                    "User-Agent": "Sir-lancebot"
+                }) as res:
+                if res.status == 200:
+                    data = await res.json()
+                    await ctx.send(data["joke"])
+                else:
+                    await ctx.send("There is no dad joke now")
```

Comment:
This path indicates something is wrong, it's probably a good idea to raise an error so it will be logged and we can look into it. `res.raise_for_status()` can be used to do this.